packages/unified-selection/src/unified-selection/HiliteSetProvider.ts

```diff
@@ -237,7 +245,13 @@ class HiliteSetProviderImpl implements HiliteSetProvider {
             SELECT ECInstanceId FROM Models
           `,
         ].join(" UNION ");
-        return from(executeQuery(this._imodelAccess, { ctes, ecsql, bindings }));
+        return from(
+          executeQuery({
+            queryExecutor: this._imodelAccess,
+            query: { ctes, ecsql, bindings },
+            config: { restartToken: `HiliteSetProvider/hilited-models-query/${Guid.createValue()}` },
```

Comment:
- let's create the id as a member of `HiliteSetProviderImpl` and use it here.
- I'd change the format of `restartToken` to `HiliteSetProvider/{componentId}/models`